src/handlers/pull_requests_assignment_update.rs

```diff
@@ -0,0 +1,68 @@
+use crate::github::retrieve_pull_requests;
+use crate::jobs::Job;
+use anyhow::Context as _;
+use async_trait::async_trait;
+use tokio_postgres::Client as DbClient;
+
+pub struct PullRequestAssignmentUpdate;
+
+#[async_trait]
+impl Job for PullRequestAssignmentUpdate {
+    fn name(&self) -> &'static str {
+        "pull_request_assignment_update"
+    }
+
+    async fn run(&self, ctx: &super::Context, _metadata: &serde_json::Value) -> anyhow::Result<()> {
+        let db = ctx.db.get().await;
+        let gh = &ctx.github;
+
+        tracing::trace!("starting pull_request_assignment_update");
+
+        // delete everything before populating
+        init_table(&db).await?;
+
+        let rust_repo = gh.repository("rust-lang/rust").await?;
+        let prs = retrieve_pull_requests(&rust_repo, &gh).await?;
+
+        // populate the table
+        for (assignees, pr_num) in prs {
+            for assignee in assignees {
+                let assignee_id = assignee.id.expect("checked");
+                ensure_team_member(&db, assignee_id, &assignee.login).await?;
+                create_team_member_workqueue(&db, assignee_id, pr_num).await?;
+            }
+        }
+
+        Ok(())
+    }
+}
+
+/// Truncate the review prefs table
+async fn init_table(db: &DbClient) -> anyhow::Result<u64> {
+    let res = db.execute("TRUNCATE review_prefs;", &[]).await?;
```

Comment:
I don't think this is right. We don't want to delete everything, just the assigned reviews.

Reply:
do you suggest to simply delete all assignments? What if we have GH users that left team? I am not sure the `users` table of the triagebot is ever purged (didn't check yet).

My reasoning is: when invoking a job to initialize the table I thought it was easier to start from a clean slate. This `init_table` should only be called when the intent is to initialize the table.

No strong opinion, though, about just deleting the assignments. It will just require adding a bit of logic to `create_team_member_workqueue` to turn it into an `upsert`

Reply:
The table is also supposed to store preferences, right? If we delete table, preferences are lost.

I guess we could periodically cleanup users, but don't think it's that big of an issue at the moment.